src/providers/trusted_service/key_management.rs

```diff
@@ -0,0 +1,149 @@
+                    insert_key_id(key_triple, key_attributes, &mut *store_handle, key_id)
+                {
+                    if self.context.destroy_key(key_id).is_err() {
+                        error!("Failed to destroy the previously generated key.");
+                    }
+                    Err(e)
+                } else {
+                    Ok(psa_generate_key::Result {})
+                }
+            }
+            Err(error) => {
+                let error = ResponseStatus::from(error);
+                format_error!("Generate key error", error);
+                Err(error)
+            }
+        }
+    }
+
+    pub(super) fn psa_import_key_internal(
+        &self,
+        app_name: ApplicationName,
+        op: psa_import_key::Operation,
+    ) -> Result<psa_import_key::Result> {
+        let key_name = op.key_name;
+        let key_attributes = op.attributes;
+        let key_data = op.data;
+        let key_triple = KeyTriple::new(app_name, ProviderID::TrustedService, key_name);
+        let mut store_handle = self
+            .key_info_store
+            .write()
+            .expect("Key store lock poisoned");
+        if key_info_exists(&key_triple, &*store_handle)? {
+            return Err(ResponseStatus::PsaErrorAlreadyExists);
+        }
+        let key_id = create_key_id(&self.id_counter)?;
+
+        match self
+            .context
+            .import_key(key_attributes, key_id, key_data.expose_secret())
+        {
+            Ok(_) => {
+                if let Err(e) =
+                    insert_key_id(key_triple, key_attributes, &mut *store_handle, key_id)
+                {
+                    if self.context.destroy_key(key_id).is_err() {
+                        error!("Failed to destroy the previously generated key.");
+                    }
+                    Err(e)
+                } else {
+                    Ok(psa_import_key::Result {})
+                }
+            }
+            Err(error) => {
+                format_error!("Import key status: ", error);
+                Err(error.into())
+            }
+        }
+    }
+
+    pub(super) fn psa_export_public_key_internal(
+        &self,
+        app_name: ApplicationName,
+        op: psa_export_public_key::Operation,
+    ) -> Result<psa_export_public_key::Result> {
+        let key_name = op.key_name;
+        let key_triple = KeyTriple::new(app_name, ProviderID::TrustedService, key_name);
+        let store_handle = self.key_info_store.read().expect("Key store lock poisoned");
+        let key_id = get_key_id(&key_triple, &*store_handle)?;
+
+        match self.context.export_public_key(key_id) {
+            Ok(pub_key) => Ok(psa_export_public_key::Result {
+                data: pub_key.into(),
+            }),
+            Err(error) => {
+                format_error!("Export key status: ", error);
+                Err(error.into())
+            }
+        }
+    }
+
+    pub(super) fn psa_destroy_key_internal(
```

Comment:
The pattern here also was changed to:
1. get the key ID from the key name using the KIM
2. destroy the key mappings
3. try to destroy the key

So that a key with the same name can be created again, even if it remains as zombie in the TS

Reply:
Dun